includes/Core/Modules/Modules.php

```diff
@@ -968,6 +972,47 @@ private function get_rest_routes() {
 					),
 				)
 			),
+			new REST_Route(
+				'core/modules/data/check-access',
+				array(
+					array(
+						'methods'             => WP_REST_Server::EDITABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$slug = $request['slug'];
+
+							try {
+								$module = $this->get_module( $slug );
+							} catch ( Exception $e ) {
+								return new WP_Error( 'invalid_module_slug', __( 'Invalid module slug.', 'google-site-kit' ), array( 'status' => 404 ) );
+							}
+
+							if ( ! $this->is_module_connected( $slug ) ) {
+								return new WP_Error( 'module_not_connected', __( 'Module is not connected.', 'google-site-kit' ), array( 'status' => 400 ) );
+							}
+
+							$access = $module->check_service_entity_access();
+
+							if ( is_wp_error( $access ) && ! is_bool( $access ) ) {
```

Comment:
Why are you checking for `is_bool` here? I would have thought that if `is_wp_error` returns `true`, `$access` is an instance of `WP_Error` and so the check for `is_bool` is irrelevant?

Reply:
Also - do we need a bit of extra logic to check whether the `WP_Error` status is `403` here? The IB mentions _not_ returning a `403` as an `WP_Error`. However I am not sure if that's relevant here, or if it relates to a permission check earlier in the call stack.

> Any exception thrown by the API request other than a 403 should be returned as a WP_Error.

Reply:
@techanvil I agree. The `is_bool` check is irrelevant since `check_service_entity_access` can only return either a `WP_Error` instance or `boolean`.
For the extra logic about checking for `403`, this is already taken care of in the individual `check_service_entity_access` methods for each module ([example](https://github.com/google/site-kit-wp/blob/ac436c9062e48befef3e9bc325de70981c175fa8/includes/Modules/Analytics.php#L1430-L1432)).